Chat name & color modal

xarantolus opened the pull request:
Hi! This is a draft for the new chat name & color modal design discussed in #2205 

Current problems I'm still working on:
* The color selection menu overflows the modal and is hidden behind it, making it impossible to select some colors
* ~~The color is sent to the server, but reopening the modal still uses the initial color (a reload or rename after color changing solves this, as then user info is resent by the websocket). This is the same behavior as before, but I'd like to fix that :) Using useState with the `displayColor` similar to the displayName doesn't seem to persist the color either.~~  The backend now re-sends user info after a color change

I'm also not entirely sure how this dialog should look when logged in e.g. via FediAuth

xarantolus committed:
Improve name & color change modal design

gabek commented:
Thanks so much for thinking through all this! I went ahead and filed a TODO #2348 to re-send client info after the user's color changed, just like when a name changes. Feel free to make that change yourself if you like, otherwise I can take care of it.

xarantolus committed:
Resend user info after color change

That way the name change dialog shows the correct
color when opening it the next time

xarantolus commented:
The backend now resends user info after a color change (ref #2348). I hope `webv2` is the correct branch for this change, the develop branch doesn't have any code for color changes. Will look at the overflowing menu next :)

xarantolus committed:
Name change modal: allow overflow of color picker

xarantolus committed:
Allow submitting form only if button is enabled

xarantolus committed:
Prettified Code!

gabek commented:
Oh awesome! And yup, `webv2` is the right place. Thanks so much for continuing to work on this.

xarantolus committed:
Make button & text input same height

gabek commented:
Thanks so much for working on this!
